v2/brigadier/src/jobs.ts

```diff
@@ -2,7 +2,66 @@ import { Event } from "./events"
 
 const defaultTimeout: number = 1000 * 60 * 15
 
-export class Job {
+/**
+ * A Brigade job.
+ * 
+ * The factory methods on this type create jobs in various ways. Jobs do not
+ * run immediately when created. You must call the Job#run method to start
+ * the job running.
+ */
+export abstract class Job {
```

Comment:
I know this would make things a little more Java-ish... but per @krancour's feedback, could we name this `AbstractJob` and rename `ContainerJob` to `Job`? That keeps naming nice and concise for regular usage, and also preserves backward compatibility. But it still would give us the lovely class hierarchy.

@adamreese would also point out that in ETL lingo, "Job" means "exactly one step in a pipeline". So in ETL, it would be a misnomer to say "A sequence job is a series of jobs..." Though `JobSequence` would be accurate.

So the terminology is "A task is made up of one or more jobs." We never really did define a concept called `Task` because we initially thought of the entire script as a task. So if we wanted to borrow the original terminiology, we _could_ make the abstract a `Task` and make `Job` one implementation, and `TaskSequence` or `SequenceTask` another.

Sorry... lots of background here.

I'll mark this comment as blocking, but I can definitely be convinced otherwise.

Reply:
Just to note the correction in chat... I totally got it backwards:

- Pipelines have jobs
- Jobs have tasks
- Task is the smallest unit of work (inside of a job)

Sorry...

Reply:
There isn't really a standard terminology it varies by tools and the layers of abstraction provided.  The abstractions should follow how each step is initialized and how data is communicated.

Another example could be:
  Task: smallest unit of work. Extract, load, or transform information. Kick off a Job
  Job: an abstract grouping of Tasks or Jobs

Reply:
There can also be an abstraction for the linkage between tasks.  Filtering, merging or any other routing of data.